src/passwordler/password_strength.py

```diff
@@ -15,17 +15,18 @@ def password_strength(password):
     Returns:
     str: a rating of either 'weak', 'good' or 'strong'
     """
+    if not isinstance(password, str):
+        raise TypeError("'password' should be of type 'string'")
+    
     count_uppercase = len(re.findall("[A-Z]", password))
     count_numbers = len(re.findall('[0-9]', password))
-    count_special_chars = len(re.findall('[!-\/:-@[-`{-~]', password)) 
+    count_special_chars = len(re.findall('[!-/:-@\\[-`{-~]', password)) 
     length = len(password)
     common_passwords = ['123456', 'password', '12345', '12345678', 'qwerty', '1234567890', '1234', 
                         'baseball', 'dragon', 'football', '1234567', 'monkey', 'letmein', 'abc123', 
                         '111111', 'mustang', 'access', 'shadow', 'master', 'michael', 'superman', 
                         '696969', '123123', 'batman', 'trustno1']
 
-    if not isinstance(password, str):
-        raise TypeError("'password' should be of type 'string'")
     if password == '':
         raise ValueError("'password' cannot be an empty string")
```

Comment:
I think we should check the Errors first (empty string in this case), then execute the main logic.

Reply:
I agree